### ISSUE TITLE
Results Page: Persist the column filters in the results table in the url 

### ISSUE THREAD

**Issue by Carla-Moz** [mozilla/perfcompare#765]
Description: As a user, when I try to filter the improvements or regressions in the results table by using the column filters, I expect the filtered view to be retained when I refresh the page. However, currently, the filters are not persisted, and the default view is restored upon refreshing.

Acceptance Criteria:

    Column filters should persist in the URL so that the exact same filtered view is retained after a page refresh.
    When a user shares the filtered view URL with others, they should see the same filtered results.
    The URL should dynamically update as filters are applied or removed.

Steps to Reproduce:
    Read the [PerfCompare Documentation](https://docs.google.com/document/d/1cpQEZXw0M5QjmNL2F1S9NKjWmz6A9Ks7zELVNTXLeB4/edit#heading=h.u62ha0d6mpxl) to get familiar with the app. 
    Navigate to the results table: [Results](https://perf.compare/compare-results?baseRev=901eda21e48867cdcc016a0dba81e51c0d4dd16b&baseRepo=mozilla-central&newRev=b48e31d47d1f562424c7693ad93f74ed39251edb&newRepo=mozilla-central&newRev=e5eef47a4db1acee5e17f5b4ea6219301d995c63&newRepo=mozilla-central&framework=13.)
    Apply filters to any columns (e.g., show only improvements or regressions).
    Refresh the page.
    Observe that the column filters are not persisted, and the default view is shown.

Expected Result: Filters applied to the columns should persist in the URL, and refreshing the page or sharing the URL should show the same filtered results.

![image](https://github.com/user-attachments/assets/5c958213-660d-4bd8-a5d9-828eff04e3eb)

![image](https://github.com/user-attachments/assets/36e3611f-d509-4f6a-8bbf-eded6541c8b7)

Hint!
Check out this closed PR to help you out. [Persist the search term in the url #724 ](https://github.com/mozilla/perfcompare/pull/724)

**Comment by syedbarimanjan** [comment]
Hi I would like to work on this.

**Comment by Carla-Moz** [comment]
@syedbarimanjan yes, go ahead and try it out! You don't need to wait to be assigned.

**Comment by Netacci** [comment]
@Carla-Moz Should this be implemented for only status and not platform and confidence? or is that a separate issue?

**Comment by Carla-Moz** [comment]
Good question! Yes, this should be implemented for status, platform, and confidence. I've updated the description.
